app/uk/gov/hmrc/apidocumentation/services/ApiDefinitionService.scala

```diff
@@ -19,27 +19,42 @@ package uk.gov.hmrc.apidocumentation.services
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
+import play.api.cache._
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.http.metrics.common._
 
 import uk.gov.hmrc.apidocumentation.connectors.ApiPlatformMicroserviceConnector
 import uk.gov.hmrc.apidocumentation.models._
+import scala.concurrent.duration._
+import uk.gov.hmrc.apidocumentation.util.ApplicationLogger
 
 trait BaseApiDefinitionService {
   def fetchExtendedDefinition(serviceName: String, developerId: Option[DeveloperIdentifier])(implicit hc: HeaderCarrier): Future[Option[ExtendedAPIDefinition]]
 
   def fetchAllDefinitions(developerId: Option[DeveloperIdentifier])(implicit hc: HeaderCarrier): Future[Seq[APIDefinition]]
 }
 
+case class PossibleDefinition(odefn: Option[ExtendedAPIDefinition])
+
 @Singleton
-class ApiDefinitionService @Inject() (val apiPlatformMicroserviceConnector: ApiPlatformMicroserviceConnector, val apiMetrics: ApiMetrics)(implicit ec: ExecutionContext)
-    extends BaseApiDefinitionService with RecordMetrics {
+class ApiDefinitionService @Inject() (cache: AsyncCacheApi, apiPlatformMicroserviceConnector: ApiPlatformMicroserviceConnector, val apiMetrics: ApiMetrics)(implicit ec: ExecutionContext)
+    extends BaseApiDefinitionService with RecordMetrics with ApplicationLogger {
   val api: API = API("api-definition")
 
-  def fetchExtendedDefinition(serviceName: String, developerId: Option[DeveloperIdentifier] = None)(implicit hc: HeaderCarrier): Future[Option[ExtendedAPIDefinition]] =
-    record {
-      apiPlatformMicroserviceConnector.fetchApiDefinition(serviceName, developerId)
+  val cacheExpiry: FiniteDuration = 5 seconds
+
+  def fetchExtendedDefinition(serviceName: String, developerId: Option[DeveloperIdentifier] = None)(implicit hc: HeaderCarrier): Future[Option[ExtendedAPIDefinition]] = {
+    val key = s"${serviceName}---${developerId.map(_.asText).getOrElse("NONE")}"
+
+    cache.getOrElseUpdate[PossibleDefinition](key, cacheExpiry) {
+      logger.info(s"Extended defintion for $serviceName for $developerId not found in cache")
```

Comment:
Speeling error.
Would be a pain if you were searching for that in Kibana.

Reply:
No - the speeling error would make it more unique :-)